Added documentation and example of W0416 Shadowed Import

sarahsonder opened the pull request:
## Motivation and Context
Since Shadowed Import is a new Pylint checker that was recently released and incorporated into PythonTA, this pull request was required to keep the PythonTA docs and examples up to date.

## Your Changes
The changes made in this pull request include adding documentation for checker Shadowed Import (W0416) and adding an example of python code that will trigger this error.

**Description**:

**Type of change** (select all that apply):

<!-- Put an `x` in all the boxes that apply. -->
<!-- Remove any lines that do not apply. -->

- [x] Documentation update (change that modifies or updates documentation only)

## Testing
I ran the example through Pylint and verified that the correct error was being raised. I also proofread my changes to the docs to check for grammatical errors.

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have verified that the CI tests have passed. <!-- (check after opening pull request) -->
- [x] I have reviewed the test coverage changes reported on Coveralls. <!-- (check after opening pull request) -->

david-yz-liu commented on docs/checkers/index.md:
```diff
@@ -1151,6 +1151,16 @@ would like to import. For example, if we have a Python file named `math.py`, cal
+
+### Shadowed import (W0416)
+
+This error occurs when a module is aliased with a name that has already been used by a previous import.
```
please reword "when a module is aliased with a name" to "when a module is imported with an aliased name"

david-yz-liu replied:
Also add a second sentence: "This prevents the original module from ever being used later in your code."

sarahsonder commented:
I think I accidentally made two commits when I only intended to make one, and I am not sure how to remove the commit. The correct changes should be in the first commit. Apologies!

david-yz-liu reviewed (commented):
@sarahsonder I left one comment. It looks like the second commit you made was empty (you can see this under the "Commits" tab), so while I'm not sure why this happened, I don't think you need to worry about it. In general don't worry about "removing" commits when you're first starting out with git, just double-check the "Files changed" diff carefully and as long as it looks okay, that should be fine. :)

sarahsonder commented:
Hi Prof. Liu, I have fixed the grammar error, but for some reason I have again made commits without meaning to. As you mentioned in your comment, I checked "git status" and "git diff" and nothing appeared out of the ordinary when I was making my commits. However, every time I tried pushing my commits, I got a message saying "Please enter a commit message to explain why this merge was necessary". My commits also kept getting rejected, which I fixed by pulling directly from my remote branch (in addition to pulling from the master branch before making my commits). Hopefully this error won't happen with future branches, but is there any way I can prevent this from happening again? Thank you!

david-yz-liu commented:
@sarahsonder see https://stackoverflow.com/questions/19085807/please-enter-a-commit-message-to-explain-why-this-merge-is-necessary-especially
